lib/elastic_apm/transport/serializers/span_serializer.rb

```diff
@@ -66,6 +66,12 @@ def build_http(http)
             }
           end
         end
+
+        private
+
+        def join_type(span)
+          [span.type, span.subtype, span.action].tap(&:compact!).join('.')
```

Comment:
Why do you call `tap` before `compact!`?

Reply:
`compact!` doesn't return `self` so I step around its return value. I used `compact!` to modify the original and skip allocating an additional array on each call. But with this I'm allocating a block instead. Don't know what's worse. I think I'll make this less fancy.

Reply:
Oh, I understand it now. The way it's written is fine with me if you'd like to keep it.